api/views/location.py

```diff
@@ -45,42 +48,68 @@ def get_queryset(self):
             return queryset
         except:
             raise ValidationError("Invalid request.")
-        
-def processQueryToDictList(query):
+
+
+class LocationCategories(generics.GenericAPIView):
+    serializer_class = LocationSerializer
+    queryset = Location.objects.all()
+
+    def post(self, request, *args, **kwargs):
+        instance = self.get_object()
+        category_ids = request.data.get('categories')
+        if category_ids:
+            categories = Category.objects.filter(id__in=category_ids)
```

Comment:
Should add some error handling, if you try to add a category that doesn't already exist, the request returns 200 but categories in the location doesn't get updated which is a bit confusing.

Reply:
what if we add a list of categories and only one doesnt exist

Reply:
hmm true, might be annoying but I'd say reject the whole request and say that the one category doesn't exist